selfdrive/boardd/boardd.cc

```diff
@@ -272,17 +372,38 @@ void panda_state_thread(bool spoofing_started) {
   // run at 2hz
   while (!do_exit && panda->connected) {
     health_t pandaState = panda->get_state();
+    health_t pandaState_aux;
+    if (panda_aux != nullptr) pandaState_aux = panda_aux->get_state();
+
+    uint8_t safety_model = pandaState.safety_model;
```

Comment:
Why copy all these fields out?

Reply:
They are used later in pandaState message if aux panda was chosen as a main "drive" panda. (But we still care about some state variables from DOS as it is driving IR, fan, camera clock etc.)

Reply:
Why not two structs and copy over the necessary values? This adds a bunch of lines that are not needed.